contracts/protocols/mainnet/liquity.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+interface TroveManagerLike {
+  function getCurrentICR(address _borrower, uint _price) external view returns (uint);
+  function getEntireDebtAndColl(address _borrower) external view returns (
+    uint debt, 
+    uint coll, 
+    uint pendingLUSDDebtReward, 
+    uint pendingETHReward
+  );
+}
+
+interface StabilityPoolLike {
+  function getCompoundedLUSDDeposit(address _depositor) external view returns (uint);
+  function getDepositorETHGain(address _depositor) external view returns (uint);
+  function getDepositorLQTYGain(address _depositor) external view returns (uint);
+}
+
+abstract contract StakingLike {
+  mapping(address => uint) public stakes;
+  function getPendingETHGain(address _user) external virtual view returns (uint);
+  function getPendingLUSDGain(address _user) external virtual view returns (uint);
+}
+
+abstract contract PriceFeedLike {
+  uint public lastGoodPrice;
+}
+
+contract Helpers {
+  TroveManagerLike internal constant troveManager =
+    TroveManagerLike(0xA39739EF8b0231DbFA0DcdA07d7e29faAbCf4bb2);
+
+  StabilityPoolLike internal constant stabilityPool = 
+    StabilityPoolLike(0x66017D22b0f8556afDd19FC67041899Eb65a21bb);
+
+  StakingLike internal constant staking =
+    StakingLike(0x4f9Fbb3f1E99B56e0Fe2892e623Ed36A76Fc605d);
+
+  PriceFeedLike internal constant priceFeed = 
+    PriceFeedLike(0x4c517D4e2C851CA76d7eC94B805269Df0f2201De);
+  
+  struct Trove {
+    uint collateral;
+    uint debt;
+    uint icr;
+  }
+
+  struct StabilityDeposit {
+    uint deposit;
+    uint ethGain;
+    uint lqtyGain;
+  }
+
+  struct Stake {
+    uint amount;
+    uint ethGain;
+    uint lusdGain;
+  }
+
+  struct Position {
+    Trove trove;
+    StabilityDeposit stability;
+    Stake stake;
+  }
+}
+
+
+contract Resolver is Helpers {
+  function getTrove(address owner) public view returns (Trove memory) {
+    (uint debt, uint collateral, uint _, uint __) = troveManager.getEntireDebtAndColl(owner);
+    uint price = priceFeed.lastGoodPrice();
```

Comment:
I wonder how important it is to use an up-to-date price here. `lastGoodPrice` can lag behind the actual Chainlink (or Tellor) price by quite a bit at times when the frequency of Liquity transactions is low.

Of course we can't just call fetchPrice() from a view function, but maybe we could create a read-only twin of the PriceFeed contract that looks at the public state of the PriceFeed and gets the price according to the same logic, but makes no state changes...

Or if that's too far fetched, we could just use a single underlying oracle (Chainlink) and read it directly.

Reply:
That’s a good question. Calling Chainlik may become less reliable in some circumstances too. What are the resulting ICR and TCR values used for?

Reply:
Great point @danielattilasimon. I guess we need to use the price that our Liquity protocol would present to the user. Creating a read only (view) `getPrice()` would simplify this significantly. 

We want to show same value a user would see inside a Liquity frontend, since this wrapper will be used by Instadapp for their frontend integration of Liquity.

Reply:
I guess one more way to go would be to get rid of the `view` modifier on `getTrove()`, which would let you call `fetchPrice()` on the `PriceFeed`. However, this may not work for Instadapp (e.g. if there's a hard requirement that all resolver functions must be constant).

Reply:
Added eth oracle price as a param: https://github.com/liquity/dsa-resolvers/pull/1/commits/34b7c390251164a64db77d24589af1620c528c41